inc/vcf/summary_report_writer.hpp

```diff
@@ -0,0 +1,107 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef VCF_SUMMARY_REPORT_WRITER_HPP
+#define VCF_SUMMARY_REPORT_WRITER_HPP
+
+#include <map>
+#include "report_writer.hpp"
+
+namespace ebi
+{
+  namespace vcf
+  {
+    class ReportVisitor : public ErrorVisitor
+    {
+      public:
+        ReportVisitor() : undefined_metadata{}, skip(false) {}
+
+        virtual void visit(Error &error) {}
+        virtual void visit(MetaSectionError &error) {}
+        virtual void visit(HeaderSectionError &error) {}
+        virtual void visit(BodySectionError &error) {}
+        virtual void visit(NoMetaDefinitionError &error)
+        {
+            if (is_bad_defined_meta(error.column, error.field)) {
+                skip = true;
+            } else {
+                add_bad_defined_meta(error.column, error.field);
+                skip = false;
+            }
+        }
+        virtual void visit(FileformatError &error) {}
+        virtual void visit(ChromosomeBodyError &error) {}
+        virtual void visit(PositionBodyError &error) {}
+        virtual void visit(IdBodyError &error) {}
+        virtual void visit(ReferenceAlleleBodyError &error) {}
+        virtual void visit(AlternateAllelesBodyError &error) {}
+        virtual void visit(QualityBodyError &error) {}
+        virtual void visit(FilterBodyError &error) {}
+        virtual void visit(InfoBodyError &error) {}
+        virtual void visit(FormatBodyError &error) {}
+        virtual void visit(SamplesBodyError &error) {}
+        virtual void visit(SamplesFieldBodyError &error) {}
+        virtual void visit(NormalizationError &error) {}
+        virtual void visit(DuplicationError &error) {}
+
+        bool should_skip_report() { return skip; }
+
+      private:
+        bool is_bad_defined_meta(std::string const & meta_type, std::string const & id) const
+        {
+            typedef std::multimap<std::string,std::string>::const_iterator iter;
+            std::pair<iter, iter> range = undefined_metadata.equal_range(meta_type);
+            for (auto & current = range.first; current != range.second; ++current) {
+                if (current->second == id) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        void add_bad_defined_meta(std::string const & meta_type, std::string const & id)
+        {
+            undefined_metadata.emplace(meta_type, id);
+        }
+
+        std::multimap<std::string, std::string> undefined_metadata;
+        bool skip;
+    };
+
+
+    class SummaryReportWriter : public ReportWriter
+    {
+      public:
+        virtual void write_error(Error &error)
+        {
+          std::cout << error.what() << std::endl;
```

Comment:
This will conflict with the plain text report, because they both write to stdout: https://github.com/jmmut/vcf-validator/blob/76ad342fc2cb16af51aa2d91fc464a7bd1354060/inc/vcf/report_writer.hpp#L42